aggregator/src/aggregator/error.rs

```diff
@@ -238,21 +238,17 @@ pub enum ReportRejectionReason {
 }
 
 impl ReportRejectionReason {
-    pub fn detail(&self) -> &'static str {
+    pub fn report_error(&self) -> ReportError {
         match self {
-            ReportRejectionReason::IntervalCollected => {
-                "Report falls into a time interval that has already been collected."
-            }
-            ReportRejectionReason::DecryptFailure => "Report share could not be decrypted.",
-            ReportRejectionReason::DecodeFailure => "Report could not be decoded.",
-            ReportRejectionReason::TaskEnded => "Task has ended.",
-            ReportRejectionReason::Expired => "Report timestamp is too old.",
-            ReportRejectionReason::TooEarly => "Report timestamp is too far in the future.",
-            ReportRejectionReason::OutdatedHpkeConfig(_) => {
-                "Report is using an outdated HPKE configuration."
-            }
-            ReportRejectionReason::TaskNotStarted => "Task has not started.",
-            ReportRejectionReason::DuplicateExtension => "Report contains duplicate extensions.",
+            ReportRejectionReason::IntervalCollected => ReportError::ReportReplayed,
+            ReportRejectionReason::DecryptFailure => ReportError::HpkeDecryptError,
+            ReportRejectionReason::DecodeFailure => ReportError::InvalidMessage,
+            ReportRejectionReason::TaskEnded => ReportError::TaskExpired,
+            ReportRejectionReason::Expired => ReportError::TaskExpired,
```

Comment:
This should use `report_dropped`, see https://www.ietf.org/archive/id/draft-ietf-ppm-dap-16.html#section-4.6.2.4-2.7.1